### PR TITLE
refactor: repo structure

### PR DESCRIPTION
**Pull request by 0xOneTony**
The restructure of the repo is described here: https://www.notion.so/defi-wonderland/Repository-Restructure-b63179e113884c79b6f797f5db940109

Also updated remappings and imports.

Decided to delete a lot of test files and test helper contracts and mocks from strategies that are in the `deprecated` folder. Things get complex if we try to have tests for old and new strategies. `v2.1` should only include tests for core and the refactored strategies, extensions and libraries.

### CODE REVIEW

File: contracts/core/Registry.sol [hunk 1]
```diff
@@ -2,17 +2,18 @@
...
 // External Libraries
-import "openzeppelin-contracts-upgradeable/contracts/access/AccessControlUpgradeable.sol";
-import "openzeppelin-contracts-upgradeable/contracts/proxy/utils/Initializable.sol";
+import {AccessControlUpgradeable} from
+    "openzeppelin-contracts-upgradeable/contracts/access/AccessControlUpgradeable.sol";
+import {Initializable} from "openzeppelin-contracts-upgradeable/contracts/proxy/utils/Initializable.sol";
 import {ERC20} from "solady/tokens/ERC20.sol";
 // Interfaces
-import "./interfaces/IRegistry.sol";
+import {IRegistry} from "./interfaces/IRegistry.sol";
```
Comment (0xAustrian) [inline contracts/core/Registry.sol:10]: (This can be in another PR) Would be cool to remove relative paths and just use the remappings.

File: test/utils/Time.sol
Comment (0xAustrian) [inline test/utils/Time.sol:1]: I think this file is not in use.

File: contracts/core/Registry.sol (hunk 1 above)
Comment (0xOneTony) [reply contracts/core/Registry.sol:10]: 100% I started changing but it was popping up more issues on certain contracts so I left for an other PR the rest. I will open a ticket

**Review by 0xAustrian — APPROVED** [review]

File: foundry.toml [hunk 2]
```diff
@@ -6,9 +6,8 @@ optimizer-runs = 1_000
 src = 'contracts'
 out = 'out'
 libs = ['node_modules', 'lib']
-test = 'test/foundry'
+test = 'test'
 cache_path = 'cache_forge'
-remappings = []
 
 # Hedgey Profile
```
Comment (ilpepepig) [inline foundry.toml:12]: The hedgey strategies tests required this, but it's no longer needed. It can be removed along with its usage here https://github.com/allo-protocol/allo-v2/blob/3ce91e28644a2f29109f2416bb5a5c8a89eed096/package.json#L11-L14

**Review by 0xAustrian — APPROVED** [review]

**Review by ilpepepig — APPROVED** [review]

**Review by thelostone-mc — APPROVED** [review]
